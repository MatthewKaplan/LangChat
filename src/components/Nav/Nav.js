import React, { Component } from 'react';
import { Text, View, StyleSheet, TouchableHighlight } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { FontAwesome } from '@expo/vector-icons';

class Nav extends Component {
	render() {
		return (
			<View style={styles.container}>
				<View style={styles.profileContainer}>
					<TouchableHighlight underlayColor="#4765A9" onPress={() => Actions.profile()}>
						<FontAwesome style={styles.profileButton} name="user" size={60} />
					</TouchableHighlight>
				</View>
		
				<TouchableHighlight
					underlayColor="#4765A9"
					onPress={() => Actions.chatRoomFrench({ language: 'french', roomId: 1, languageId: 1 })}
				>
					<Text Text style={styles.button}>
						chatroom french
					</Text>
				</TouchableHighlight>
				<TouchableHighlight
					underlayColor="#4765A9"
					onPress={() => Actions.chatRoomSpanish({ language: 'spanish', roomId: 2, languageId: 2 })}
				>
					<Text Text style={styles.button}>
						chatroom spanish
					</Text>
				</TouchableHighlight>
			
					<TouchableHighlight underlayColor="#4765A9" onPress={() => Actions.tutorial()}>
					<Text Text style={styles.button}>
						tutorial
					</Text>
				</TouchableHighlight>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1.0,
		marginTop: 5,
		color: 'white'
	},
	button: {
		color: '#E8EAEF',
		width: '100%',
		textAlign: 'left',
		padding: 10,
		fontSize: 20
	},
	buttonTitle: {
		fontSize: 15,
		color: 'white'
	},
	profileContainer: {
		marginTop: 10
	},
	profileButton : {
		color: '#E8EAEF',
		width: '100%',
		textAlign: 'left',
		padding: 10
	}
});

export default Nav;