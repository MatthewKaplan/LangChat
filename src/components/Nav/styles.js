import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	container: {
		flex: 1.0,
		marginTop: 5,
		color: 'white'
	},
	profileIcon: {
		marginLeft: 10
	},
	button: {
		width: '100%',
		height: 40,
		justifyContent: 'flex-start',
		paddingLeft: 10,
		alignItems: 'center',
		flexDirection: 'row'
	},
	buttonTitle: {
		fontSize: 18,
		color: '#fff',
		paddingLeft: 10
	},
	profileContainer: {
		marginTop: 10
	},
	profileButton: {
		color: '#E8EAEF',
		width: '100%',
		textAlign: 'left',
		padding: 10,
		marginRight: 10
	},
	navHeader: {
		color: 'grey',
		padding: 10,
		fontSize: 20,
		borderBottomColor: 'grey',
		borderBottomWidth: 2
	}
});
