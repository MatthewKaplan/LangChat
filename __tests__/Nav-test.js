// __tests__/Nav-test.js
import React from 'react';
import Nav from '../src/components/Nav/Nav';

import renderer from 'react-test-renderer';

describe('Nav Component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<Nav />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
