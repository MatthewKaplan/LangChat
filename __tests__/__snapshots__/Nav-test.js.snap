// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component should match snapshot 1`] = `
<View
  style={
    Object {
      "color": "white",
      "flex": 1,
      "marginTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      splash
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      login
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      signup
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      chatroom french
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      chatroom spanish
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      tutorial
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      Text={true}
      style={
        Object {
          "color": "#E8EAEF",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      tutorial
    </Text>
  </View>
</View>
`;
